lib/Conversion/ImportVerilog/TimingControls.cpp

```diff
@@ -0,0 +1,62 @@
+//===- TimingControl.cpp - Slang timing control conversion ----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImportVerilogInternals.h"
+#include "slang/ast/TimingControl.h"
+using namespace circt;
+using namespace ImportVerilog;
+namespace {
+struct TimingCtrlVisitor {
+  Context &context;
+  Location loc;
+  OpBuilder &builder;
+
+  TimingCtrlVisitor(Context &context, Location loc)
+      : context(context), loc(loc), builder(context.builder) {}
+
+  LogicalResult visit(const slang::ast::SignalEventControl &ctrl) {
+    auto loc = context.convertLocation(ctrl.sourceRange.start());
+    auto input = context.convertRvalueExpression(ctrl.expr);
+    builder.create<moore::EventOp>(loc, static_cast<moore::Edge>(ctrl.edge),
+                                   input);
+    return success();
+  }
+
+  LogicalResult visit(const slang::ast::ImplicitEventControl &ctrl) {
+    return success();
+  }
+
+  LogicalResult visit(const slang::ast::EventListControl &ctrl) {
+    for (auto *event : ctrl.as<slang::ast::EventListControl>().events) {
+      if (failed(context.convertTimingControl(*event)))
+        return failure();
+    }
+    return success();
+  }
```

Comment:
Could we merge this into `Statements.cpp` into the statement visitor? If I remember correctly, event and delay control are always associated with a statement, so it would make sense to just have that there.

Reply:
Yeah, you're right! However, due to slang separating `timing controls` and `statements` into different classes, the former belongs to `slang::ast::TimingControl`, and the latter is `slang::ast::Statement`. Sure, the `TimedStatement` inherits from `Statement`, but we need to go via `stmt.timing` to visit the `timing control list`